service/worker/scanner/executions/interfaces.go

```diff
@@ -24,24 +24,36 @@
 
 package executions
 
-import "go.temporal.io/server/service/worker/scanner/executor"
-
-type handlerStatus = executor.TaskStatus
-
-const (
-	handlerStatusDone  = executor.TaskStatusDone
-	handlerStatusErr   = executor.TaskStatusErr
-	handlerStatusDefer = executor.TaskStatusDefer
+import (
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/persistence"
 )
 
-const scannerTaskQueuePrefix = "temporal-sys-executions-scanner"
+type (
+	// ExecutionData wraps data on execution that is suitable for running validations.
+	// Extend this structure if more data is required for new checks.
+	ExecutionData struct {
+		mutableState  *persistencespb.WorkflowMutableState
+		history       *persistence.ReadHistoryBranchResponse
```

Comment:
take a look at this code:
https://github.com/temporalio/temporal/blob/b88ecb124f347f4ab35fa8bcc334a7e1707815cf/service/history/workflowResetter.go#L539-L548
https://github.com/temporalio/temporal/blob/b88ecb124f347f4ab35fa8bcc334a7e1707815cf/service/history/workflowResetter.go#L582-L608

Reply:
basically keeping the iterator, not the response